### PR TITLE
loopback: Utimens: enable Nsec precision for dates after 1970

### PR DESCRIPTION
**Pull request by rfjakob**
syscall.NsecToTimespec is broken for dates before 1970 but works
fine otherwise.

Fix #100 as well and add a testcase.

### CODE REVIEW

File: fuse/nodefs/files_linux.go [hunk 1]
```diff
@@ -32,14 +32,19 @@ func (f *loopbackFile) Utimens(a *time.Time, m *time.Time) fuse.Status {
...
+			ts[0].Nsec = 0
+		}
 	}
 
 	if m == nil {
 		ts[1].Nsec = _UTIME_OMIT
 	} else {
-		ts[1] = syscall.NsecToTimespec(a.UnixNano())
-		ts[1].Nsec = 0
+		ts[1] = syscall.NsecToTimespec(m.UnixNano())
```
Comment (hanwen) [inline fuse/nodefs/files_linux.go]: maybe have our own nsecToTimespec() that does this just once?

**Comment by rfjakob** [comment]
Done in the second commit: https://github.com/hanwen/go-fuse/pull/131/commits/6e43fd0f73de2235e5168b67f49f8882aa19ea32

**Comment by rfjakob** [comment]
Ping? The typo in [files_linux.go#L41](https://github.com/hanwen/go-fuse/blob/master/fuse/nodefs/files_linux.go#L41) causes crashes in gocryptfs ( https://github.com/rfjakob/gocryptfs/issues/48 ). This is also fixed in this PR.

**Comment by rfjakob** [comment]
Hi @hanwen , is there anything else I can do to get this merged?

**Comment by hanwen** [comment]
sorry, this went off my radar. Can you squash the commit, and drop the fuse.test binary from it?

**Comment by rfjakob** [comment]
Squashed and updated. Sorry about that binary! It's gone now as well.

**Pull request merged**